search/search_quality/assessment_tool/result_view.cpp

```diff
@@ -0,0 +1,97 @@
+#include "search/search_quality/assessment_tool/result_view.hpp"
+
+#include "search/result.hpp"
+#include "search/search_quality/assessment_tool/helpers.hpp"
+
+#include <string>
+
+#include <QtWidgets/QHBoxLayout>
+#include <QtWidgets/QLabel>
+#include <QtWidgets/QRadioButton>
+#include <QtWidgets/QVBoxLayout>
+
+namespace
+{
+QLabel * CreateLabel(QWidget & parent)
+{
+  QLabel * label = new QLabel(&parent);
+  label->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Minimum);
+  label->setWordWrap(true);
+  return label;
+}
+
+void SetText(QLabel & label, std::string const & text) {
+  if (text.empty())
+  {
+    label.hide();
+    return;
+  }
+
+  label.setText(ToQString(text));
+  label.show();
+}
+}  // namespace
+
+ResultView::ResultView(search::Result const & result, QWidget & parent) : QWidget(&parent)
+{
+  Init();
+  SetContents(result);
+}
+
+void ResultView::SetRelevance(search::Sample::Result::Relevance relevance)
+{
+  m_irrelevant->setChecked(false);
+  m_relevant->setChecked(false);
+  m_vital->setChecked(false);
+  switch (relevance)
+  {
+  case search::Sample::Result::RELEVANCE_IRRELEVANT: m_irrelevant->setChecked(true); break;
+  case search::Sample::Result::RELEVANCE_RELEVANT: m_relevant->setChecked(true); break;
+  case search::Sample::Result::RELEVANCE_VITAL: m_vital->setChecked(true); break;
+  }
+}
+
+void ResultView::Init()
+{
+  auto * layout = new QVBoxLayout(this /* parent */);
+  layout->setSizeConstraint(QLayout::SetMinimumSize);
+  setLayout(layout);
+
+  m_string = CreateLabel(*this /* parent */);
+  layout->addWidget(m_string);
+
+  m_type = CreateLabel(*this /* parent */);
+  m_type->setStyleSheet("QLabel { font-size : 8pt }");
+  layout->addWidget(m_type);
+
+  m_address = CreateLabel(*this /* parent */);
+  m_address->setStyleSheet("QLabel { color : green ; font-size : 8pt }");
+  layout->addWidget(m_address);
+
+  {
+    auto * group = new QWidget(this /* parent */);
+    auto * groupLayout = new QHBoxLayout(group /* parent */);
+    group->setLayout(groupLayout);
+
+    m_irrelevant = new QRadioButton("0", this /* parent */);
+    m_relevant = new QRadioButton("+1", this /* parent */);
+    m_vital = new QRadioButton("+2", this /* parent */);
```

Comment:
Numbers are shorter than text, moreover, we're going to replace them by icons. So, let's keep this as is.